packages/uma-auth-client/src/utils/defineWebComponent.tsx

```diff
@@ -15,18 +16,34 @@ export default function defineWebComponent(
 
       constructor() {
         super();
-        this.root = ReactDOM.createRoot(this);
+        this.attachShadow({ mode: "open" });
+        this.root = ReactDOM.createRoot(this.shadowRoot as ShadowRoot);
       }
 
       connectedCallback() {
-        this.root.render(
+        // Create insertion node for Emotion
+        const emotionInsertionPoint = document.createElement("meta");
+        emotionInsertionPoint.setAttribute("name", "emotion-insertion-point");
+        this.shadowRoot?.appendChild(emotionInsertionPoint);
+
+        // Create emotion cache so dynamic styles get defined to the insertion node
+        const cache = createCache({
+          key: "uma-auth-client",
```

Comment:
I think so https://emotion.sh/docs/@emotion/cache#key

but since each instance would be in its own shadow root the styles don't affect each other despite having the same class names. Here's a test showing it working:

![Screenshot 2024-08-13 at 1.49.26 PM.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/NU8OmLauzLqa61yWDJkY/844ab793-70e4-4616-b857-666d3dbc4f42.png)